### ISSUE TITLE
Wrong TypeScript definition on Unit division

### ISSUE THREAD

**Issue by hw762** [josdejong/mathjs#2185]
I don't know if I should file bug report on typing here, but here is the case:

```typescript
const a: Unit = unit("1 km")
const b: Unit = unit("1 m")
const ratio: Unit = divide(a, b)
```

This type checks, but `ratio` is actually `number` at runtime. This causes problem as I cannot write correct code without type assertion (i.e. `as unknown as number`).

**Comment by josdejong** [comment]
Thanks for reporting @hw762. It looks like the TypeScript definitions are not entirely accurate in this case, a division of two units having the same type results in a `number` indeed.

Anyone able to update the TypeScript definitions? Help would be very welcome.

**Comment by CatsMiaow** [comment]
With a similar problem, `pickRandom` can handle strings as well, but only the number type is defined, resulting in a type error.
```ts
math.pickRandom(['a', 'b', 'c', 'd'], 2); // ["d", "b"]
```
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bb8745c9b3d5792a3783a9941c1fdff4ef8cf140/types/mathjs/index.d.ts#L1619-L1623

In my opinion, managing types in DefinitelyTyped is less accessible to contributions.
It would be nice to manage it within the repository like [this](https://github.com/iamkun/dayjs/blob/master/types/index.d.ts).

**Comment by josdejong** [comment]
> In my opinion, managing types in DefinitelyTyped is less accessible to contributions.

Yeah you're right. I've been a bit reluctant adding TypeScript definitions into the library but I think it makes sense, TypeScript being the de-facto standard in this regard. See also #2076, going a step further ;) .

Who is willing to move the type definitions from DefinitelyTyped into mathjs?

**Comment by josdejong** [comment]
I created a separate issue for your idea @CatsMiaow : #2187

**Comment by josdejong** [comment]
Fixed now in `mathjs@9.4.0` with the now build-in type definitions
